handlers/api.go

```diff
@@ -1522,7 +1537,34 @@ func ApiValidatorDailyStats(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	rows, err := db.ReaderDb.Query(`
+	result := []struct {
```

Comment:
All this and the block starting at line 1601
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/a8c4a970428a6ccc3ead173ce7a7a2d8412a629e/handlers/api.go#L1601

... this is SO much duplicated code and reduces readability a lot. I see where you are coming from, but consider the following approach:

Do everything like before but instead of calling `returnQueryResultsAsArray` at the end, call
```
data, err := utils.SqlRowsToJSON(rows)

if err != nil {
	sendErrorResponse(w, r.URL.String(), "could not parse db results")
	return
}
```
directly and add your new fields (`DayStart` and `DayEnd`) manually.
Then proceed with `sendOKResponse`.

Reply:
I rewrote `returnQueryResults` and `returnQueryResultsAsArray` to take functions as parameters (**adjustQueryEntriesFuncs**)that adjust the query results.
This seems future proof to me and avoids the most duplicate code.

Please let me know what you think of this approach.